### DISCUSSION TITLE
Showing a tooltip on repeatable elements like table cells

### DISCUSSION THREAD

**Discussion by emadabdulrahim** [floating-ui/floating-ui#2921] [Q&A] [answered]
I'm trying to figure out the best approach for showing a tooltip on repeatable elements, such as table cells, using the floating-ui library.

It makes sense to render only one floating UI element representing the tooltip content. However, I'm unsure how to wire up the ref of the cell that was interacted with.
Here are a few questions I have:

1. Should I manage a ref state, set it to the cell that was clicked on, and have a ﻿`useEffect` hook that triggers when the cell ref state changes to call ﻿`refs.setReference(cellRef)`?
2. Are there any other recommended approaches for handling this scenario? 

I would greatly appreciate any guidance or examples on the proper way to handle this.

Thanks!

**Comment by atomiks** [comment, marked as answer]
This is a "singleton" tooltip pattern, and there's no official support with the native interaction hooks right now, assuming you're using `@floating-ui/react`.

But yeah it's possible to build it manually by calling `refs.setReference(cellNode)` on whatever cell is being hovered or was clicked to make the reusable tooltip element anchor to it. Of course, you need to handle opening/closing it as well in this scenario.

```js
<Cell onMouseEnter={({currentTarget}) => refs.setReference(currentTarget)} />
```

**Reply by emadabdulrahim** [reply]
@atomiks Quick question, so I ended up implementing what I had planned and it works well. There's an issue where the table stops scrolling when the mouse is hover the portaled floating element. Do you know how can I make scrolling event continue to bubble up to the table, without using something like `pointer-events: none` on the floating element, since I still want it to be interactive (clickable, selectable text, etc) 

My setup looks like this 

```tsx
// where it's rendered

<TableStoreProvider<TData>>
        <Table.Container table={table} variant={variant} loading={loading} {...props}>
          <Table.Header table={table} bg={headerColor} />
          <Table.Body
            table={table}
            isStriped={stripedRows || table.getState().stripedRows}
            loading={loading}
          />
        </Table.Container>
        {!loading && !tableUtils.isTableResizing(table) && (
          <FloatingCell cellSize={table.getState().size} variant={variant} />
        )}
      </TableStoreProvider>
```

```tsx
// inside FloatingCell
   const [isOpen, setIsOpen] = React.useState(false);

  const { floatingStyles, refs, context } = useFloating<HTMLInputElement>({
    open: isOpen,
    onOpenChange: setIsOpen,
    whileElementsMounted: autoUpdate,
    placement: "bottom-start",
    middleware: [
      offset(({ rects, elements }) => {
        return {
          mainAxis: -rects.reference.height,
          // offset for cell border from spreadsheet variant
          crossAxis: variant === "spreadsheet" ? 1 : 0,
        };
      }),
      flip(),
      shift(),
      size({
        apply({ elements, rects }) {
          Object.assign(elements.floating.style, {
            minWidth: `${rects.reference.width}px`,
            minHeight: `${rects.reference.height}px`,
          });
        },
      }),
    ],
  });
  
  ...
  const hover = useHover(context, {
    restMs: 120,
    handleClose: safePolygon({
      buffer: 20,
    }),
  });
  const dismiss = useDismiss(context);

  const { getReferenceProps, getFloatingProps } = useInteractions([hover, dismiss]);
  
  ...
  return (
   <FloatingPortal>
      {isOpen && cell.node && (
        <div
          {...getFloatingProps}
          ref={refs.setFloating}
          className={styles.overlay}
          data-size={cellSize}
          style={floatingStyles}
        >
          {tableUtils.renderCell(cell.object!, true)}
        </div>
      )}
    </FloatingPortal>
    )
 ```
 
 
 Also happy to post my solution for other folks to benefit from
 

https://github.com/floating-ui/floating-ui/assets/7950797/d304b400-f291-471f-83c1-e01b7896c741

**Reply by emadabdulrahim** [reply]
Hmm I think I actually found the issue. Once I moved the floating element inside the scrollable container, and not mounting it on a portal, I can scroll the container even when the mouse is over the floating element.
